docs/changes/table/11585.bugfix.rst

```diff
@@ -0,0 +1,3 @@
+Fix a bug where a string-valued ``Column`` that happened to have a ``unit``
```

Comment:
This actually applies to `object` type as well. It should also apply to `bool` which is not numeric (at least from https://numpy.org/doc/stable/reference/arrays.scalars.html), but instead I see:
```
In [2]: c = Column([1,0], dtype=bool)

In [3]: c.unit = 'blah'

In [4]: t = QTable([c])

In [5]: t
Out[5]: 
<QTable length=2>
 col0
 blah
 bool
-----
 True
False

In [6]: type(t['col0'])
Out[6]: astropy.units.quantity.Quantity
```
To me this is a bug on Quantity.

Reply:
Hmm, yes, agreed that is not very logical. But separate from this PR...